File: apps/motoboy/app/src/main/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/DeliveryDisplayCode.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

/**
 * Mostra o numero curto do dia quando a API ja trouxe a sequencia da loja.
 */
fun deliveryDailyNumberLabel(storeDailyNumber: Int?): String? {
  val number = storeDailyNumber ?: return null
  return "N. dia ${number.toString().padStart(3, '0')}"
}

/**
 * Mantem o codigo publico completo como fallback para entregas antigas sem numeracao diaria.
 */
fun deliveryPrimaryCodeLabel(publicCode: String, storeDailyNumber: Int?): String {
  return deliveryDailyNumberLabel(storeDailyNumber) ?: publicCode
}

File: apps/motoboy-pwa/src/deliveryDisplayCode.ts
type DeliveryDisplayCodeInput = {
  publicCode: string;
  storeDailyNumber?: number | null;
};

/** Mostra o numero curto do dia quando a API ja trouxe a sequencia da loja. */
export function deliveryDailyNumberLabel(delivery: DeliveryDisplayCodeInput) {
  return delivery.storeDailyNumber ? `N. dia ${String(delivery.storeDailyNumber).padStart(3, "0")}` : null;
}

/** Mantem o codigo publico completo como fallback para entregas antigas sem sequencia diaria. */
export function deliveryPrimaryCodeLabel(delivery: DeliveryDisplayCodeInput) {
  return deliveryDailyNumberLabel(delivery) ?? delivery.publicCode;
}
